src/cunumeric/index/repeat_omp.cc

```diff
@@ -0,0 +1,95 @@
+/* Copyright 2021-2022 NVIDIA Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include "cunumeric/index/repeat.h"
+#include "cunumeric/index/repeat_template.inl"
+#include <omp.h>
+
+namespace cunumeric {
+
+using namespace Legion;
+using namespace legate;
+
+template <LegateTypeCode CODE, int DIM>
+struct RepeatImplBody<VariantKind::OMP, CODE, DIM> {
+  using VAL = legate_type_of<CODE>;
+
+  size_t operator()(Buffer<VAL>& out,
+                    const AccessorRO<VAL, DIM>& in,
+                    const int64_t repeats,
+                    const int32_t axis,
+                    const Pitches<DIM - 1>& pitches,
+                    const Rect<DIM>& rect) const
+  {
+    const size_t volume = rect.volume();
+    size_t size         = volume * repeats;
+
+    out             = create_buffer<VAL>(size, Memory::Kind::SYSTEM_MEM);
+    int64_t out_idx = 0;
+#pragma omp parallel for schedule(static)
+    for (size_t idx = 0; idx < size; ++idx) {
+      size_t p_idx = idx / repeats;
+      auto p       = pitches.unflatten(p_idx, rect.lo);
+      out[idx]     = in[p];
+    }
+    return size;
+  }
+
+  size_t operator()(Buffer<VAL>& out,
+                    const AccessorRO<VAL, DIM>& in,
+                    const AccessorRO<int64_t, DIM>& repeats,
+                    const int32_t axis,
+                    const Pitches<DIM - 1>& pitches,
+                    const Rect<DIM>& rect) const
+  {
+    const size_t volume = rect.volume();
+    int64_t size        = 0;
+    int64_t* offsets    = static_cast<int64_t*>(alloca(volume * sizeof(int64_t)));
```

Comment:
I suggest you use this thread local storage instead: https://github.com/nv-legate/cunumeric/blob/branch-22.03/src/cunumeric/omp_help.h. `alloca` was causing issues in other OpenMP tasks.

Reply:
I think I just need to replace it with `malloc` since this is not a thread-local storage (its size is `volume` which may be different from `num_threads`)

Reply:
if that's the case, why didn't you use deferred buffers? any resource usage outside the runtime's management is not good, especially for things this big.